src/push/Permission.js

```diff
@@ -62,13 +66,14 @@ export default class Permission {
             this._win.Notification &&
             this._win.Notification.requestPermission
         ) {
-            /* Chrome 23+ */
-            this._win.Notification
-                .requestPermission()
-                .then(resolve)
-                .catch(function() {
+            /* Safari 12+ */
+            request = this._win.Notification.requestPermission(resolve);
+            if (request && request.then) {
```

Comment:
I'm a little confused here.. this will prevent the console from erroring, but what is the fallback if the request variable proves to not be a promise? Does it just do nothing?

Reply:
If the request variable is not a promise, we are in Safari 12. In this version of Safari, the result of `requestPermission` is `undefined`. In Chrome 23+, the result of `requestPermission` is a Promise.

Reply:
See reference links in pull request body.

Reply:
Ah, I see.. alright that makes sense. I take it then in Chrome the passed `resolve` argument to `requestPermission()` is just ignored? Would you mind just adding some comments elaborating on the logic here just so it's easier to understand at first glance? More specifically, probably just mentioning how that argument will be used in Safari but not Chrome, if that makes sense.

Reply:
Yes, in Chrome, the unused argument to `requestPermission()` is ignored.

I'll add comments! :)